### DISCUSSION TITLE
Poor performance on iOS simulator (iOS 14.2)

### DISCUSSION THREAD

**Discussion by valdrinkoshi** [google/model-viewer#3376] [Q&A] [answered]
### Description
Poor performance on iOS simulator (my machine: MacBook Pro (15-inch, 2019)  2.6 GHz 6-Core Intel Core i7, 32 GB 2400 MHz DDR4), see recording 

https://user-images.githubusercontent.com/6173664/163832054-0b2c9f60-bfdd-41ac-bac8-9a02ff833355.mov

From the "timelines" tab, it looks like the culprit is `drawElements` 

<img width="952" alt="Screen Shot 2022-04-18 at 11 22 52 AM" src="https://user-images.githubusercontent.com/6173664/163832418-0c12578e-f37a-4083-a5b0-140f6566236a.png">

Below the captures of the "timelines" and "graphics" tabs: 

[modelviewer.dev_ios_simulator_safari_captures.zip](https://github.com/google/model-viewer/files/8506601/modelviewer.dev_ios_simulator_safari_captures.zip)

#### Live Demo
https://modelviewer.dev

### Version
prod (the one used by modelviewer.dev)

### Browser Affected

- [ ] Chrome, version: xx.x.xxxx.xx
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [x] Safari

### OS

- [ ] Android
- [x] iOS
- [ ] Linux
- [x] MacOS
- [ ] Windows

**Comment by elalish** [comment, marked as answer]
I think iOS simulator is doing a very poor job of emulating a GPU. Is iOS 15 any better? And do other WebGL pages run any better? This definitely doesn't look like a MV bug to me.

**Reply by valdrinkoshi** [reply]
Ah, iOS 15 is way better indeed!

<img width="1542" alt="Screen Shot 2022-04-18 at 2 33 43 PM" src="https://user-images.githubusercontent.com/6173664/163856872-5513e5a2-3d3b-4869-9802-2a779fc992cb.png">

I was stuck at iOS 14.2 with the iOS simulator from an XCode Beta release 🤦‍♂️
